src/api/docker-compose.yml

```diff
@@ -106,3 +106,52 @@ services:
       - 'traefik.http.routers.login.rule=Host(`login.localhost`)'
       # Specify the login port
       - 'traefik.http.services.login.loadbalancer.server.port=8080'
+      # Add middleware to this route to strip the /v1/posts prefix
+      - 'traefik.http.middlewares.strip_api_prefix.stripprefix.prefixes=/${API_VERSION}/image'
+      - 'traefik.http.middlewares.strip_api_prefix.stripprefix.forceSlash=true'
+      - 'traefik.http.routers.image.middlewares=strip_api_prefix'
+
+  redis:
+    image: redis:latest
+    container_name: api_redis
+    ports:
+      - ${REDIS_PORT}
+    command: ['redis-server', '--appendonly', 'yes']
+    volumes:
+      - ../../redis-data:/data
+
+  elasticsearch:
+    image: docker.elastic.co/elasticsearch/elasticsearch:7.9.3
+    environment:
+      - bootstrap.memory_lock=true
+      - 'ES_JAVA_OPTS=-Xms512m -Xmx512m'
+      - discovery.type=single-node
+    ports:
+      - ${ELASTIC_PORT}
+
+  # posts service
+  posts:
+    container_name: 'posts'
+    build:
+      context: ./posts
+      dockerfile: Dockerfile
+    environment:
+      - POSTS_PORT
+      - REDIS_URL
+      - ELASTIC_URL
+    ports:
+      - ${POSTS_PORT}
+    depends_on:
+      - traefik
+      - redis
+    labels:
+      # Enable Traefik for the posts service container
+      - 'traefik.enable=true'
+      # Enables routing for the posts service at /v1/posts
+      - 'traefik.http.routers.posts.rule=Host(`${API_HOST}`) && PathPrefix(`/${API_VERSION}/posts`)'
+      # Specifiy the posts service port
+      - 'traefik.http.services.posts.loadbalancer.server.port=${POSTS_PORT}'
+      # Add middleware to this route to strip the /v1/posts prefix and use gzip compression
+      - 'traefik.http.middlewares.strip_api_prefix.stripprefix.prefixes=/${API_VERSION}/posts'
```

Comment:
I need to research this more, but I don't think you can reuse this name `strip_api_prefix` again, since it's been used on the image service.  Probably it needs to be `strip_image_prefix` and `strip_posts_prefix` etc.

Reply:
This might have been the root cause of the traefik http routing issue I had whenever I would run both image and posts services together. I'll test and resolve if this works